### ISSUE TITLE
Call to a member function has() on null when calling /contact

### ISSUE THREAD

**Issue by AlainT** [975L/ContactFormBundle#1]
Hello,
Symfony 3.4.12 and Php 7.1.10, i've just installed and configured your bundle.
But i've an issue "Call to a member function has()" when trying to display the form.
Debub info:
```
in vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Controller/ControllerTrait.php (line 444)

         *
         * @final since version 3.4
         */
        protected function getUser()
        {
            if (!$this->container->has('security.token_storage')) {
                throw new \LogicException('The SecurityBundle is not registered in your application. Try running "composer require symfony/security-bundle".');
            }
```

**Comment by Laurent3170** [comment]
Hi, it seems that it requires the security-bundle, but to be sure and more helpful, I need to have the debug from the ContactFormBundle, the file and line where the error occurs.

**Comment by AlainT** [comment]
bonjour, je n'avais pas vu que vous étiez français :-)
une log plus complète :
```
in vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Controller/ControllerTrait.php (line 444)

         *
         * @final since version 3.4
         */
        protected function getUser()
        {
            if (!$this->container->has('security.token_storage')) {
                throw new \LogicException('The SecurityBundle is not registered in your application. Try running "composer require symfony/security-bundle".');
            }
            if (null === $token = $this->container->get('security.token_storage')->getToken()) {
                return;

Controller->getUser() in vendor/c975l/contactform-bundle/Controller/ContactFormController.php (line 40)

            //Defines the referer to redirect to after submission
            $session = $request->getSession();
            $session->set('redirectUrl', $request->headers->get('referer'));
            //Gets email and name if user is logged
            $user = $this->getUser();
            if ($user !== null) {
                $userEmail = $user->getEmail();
                $name = $user->getFirstname();
                $name .= $name != '' ? ' ' : '';
                $name .= $user->getLastname();

ContactFormController->displayAction(object(Request)) in vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php (line 151)

            $this->dispatcher->dispatch(KernelEvents::CONTROLLER_ARGUMENTS, $event);
            $controller = $event->getController();
            $arguments = $event->getArguments();
            // call controller
            $response = \call_user_func_array($controller, $arguments);
            // view
            if (!$response instanceof Response) {
                $event = new GetResponseForControllerResultEvent($this, $request, $type, $response);
                $this->dispatcher->dispatch(KernelEvents::VIEW, $event);

```
un problème à la récupération du user ?
[dev.log](https://github.com/975L/ContactFormBundle/files/2209792/dev.log)
 je précise que je suis loggué, authentifié (j'utilise FOSUserBundle),

**Comment by Laurent3170** [comment]
Oui français ;-)
But I will continue in english for other people reading the issue.

So, yes it seems that the user cannot be get, due to the non-registration of the security bundle. Try to register it (composer and AppKernel.php) to see if it works, as the method to get the user is well called by `$this->getUser()`.

**Comment by AlainT** [comment]
hello,
the security bundle is registered, wrong way...
i've found these links:
https://github.com/schmittjoh/JMSDiExtraBundle/issues/307
https://github.com/schmittjoh/JMSDiExtraBundle/issues/282
https://stackoverflow.com/questions/48153155/vendor-updates-broke-fos-user-bundle-with-call-to-a-member-function-has-on-a
...
may be incompatible with FOSUserBundle ?

**Comment by Laurent3170** [comment]
I don’t see why it would be incompatible with FOSUserBundle, I’ve been using it before developing my own bundle and it was working fine.
Furthermore, the way to call the user `$this->getUser()` is not relying on FOSUserBundle but it’s Symfony related, via the security bundle.
I can’t see how to reproduce the error and have no time before saturday to do so.

**Comment by AlainT** [comment]
Ok, don't worry. The bug seems effectively relative to another bundle and Symfony core...
https://github.com/symfony/symfony/issues/27053
i've unregistered JMSDiExtraBundle, added firstname and lastname to my user class, installed your c975l/includelibrary-bundle with seems to be a dependency for inc_lib calls.
The form behind /contact is displaying, but without any styles...
Still working on,

**Comment by AlainT** [comment]
Issue with included bootstrap, no Jquery found it seems,
![capture d ecran - 20072018 - 11 18 15](https://user-images.githubusercontent.com/451803/42994741-3c418d1a-8c0f-11e8-9d1e-2b2d451c36d9.png)

**Comment by AlainT** [comment]
Other thing : 

- "gpdr" if false by config, but the checkbox is present. 
- Lastname and firstname are null in my case (username is not null), so i think normal that editbox "Nom" is blank. but what about email ? email and email_canonical are not null.

**Comment by Laurent3170** [comment]
Yes you are right, there are some errors. I will attempt to correct them tomorrow. Thanks for having letting me know.

Include-library is declared in the composer and should be installed by default, as, you’re right, it’s used for `inc_lib`.

**Comment by AlainT** [comment]
you're welcome

**Comment by Laurent3170** [comment]
Everything should be solved by PR #2. I have also released v1.8.3.
I will refactor Controller to be more SOLID compliant and release a v1.9 later. It will also include the use of lastname, etc.
